src/Foundation/NSDate.cs

```diff
@@ -29,56 +29,66 @@
 using System.Reflection;
 using System.Collections.Generic;
 using System.Runtime.InteropServices;
-
+using System.Threading;
 using ObjCRuntime;
 
 namespace Foundation {
 
 	public partial class NSDate {
+		const double NANOSECS_PER_MILLISEC = 1000000.0;
 		const long NSDATE_TICKS = 631139040000000000;
 
+		private static readonly NSCalendar calendar = new NSCalendar (NSCalendarType.Gregorian) { TimeZone = NSTimeZone.FromName ("UTC") };
+		private static readonly ThreadLocal<NSDateComponents> threadComponents = new ThreadLocal<NSDateComponents> (() => new NSDateComponents ());
+
 #if XAMCORE_2_0
 		// now explicit since data can be lost for small/large values of DateTime
-		public static explicit operator DateTime (NSDate d)
-		{
+		public static explicit operator DateTime (NSDate d) {
 			double secs = d.SecondsSinceReferenceDate;
-			if ((secs < -63113904000) || (secs > 252423993599))
-				throw new ArgumentOutOfRangeException ("Value is outside the range of NSDate");
 
-			return new DateTime ((long)(secs * TimeSpan.TicksPerSecond + NSDATE_TICKS), DateTimeKind.Utc);
+			if (IntPtr.Size == 4) {
```

Comment:
Can you please add an explanation on why we have a different code path for the 32b ones? At some point someone is going to see this and decide to remove it.

Reply:
that looks like the original logic, right ?
also curious why